lib/spack/spack/repo.py

```diff
@@ -2093,10 +2099,12 @@ def __init__(self, name, repo=None):
 
                 # We need to compare the base package name
                 pkg_name = name.rsplit(".", 1)[-1]
-                try:
-                    similar = difflib.get_close_matches(pkg_name, repo.all_package_names())
-                except Exception:
-                    similar = []
+                similar = []
+                if isinstance(repo, (RepoPath, llnl.util.lang.Singleton)):
```

Comment:
I would use `not isinstance(repo, <complement>)` then. Singleton shouldn't be used here.

Reply:
It did bother me to have to add that. Mypy certainly didn't complain about that type being missing when I had a tight hint.

Reply:
Used `ensure_unwrapped` instead.